app/views/course/assessment/question/multiple_responses/_multiple_response_details.json.jbuilder

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+json.partial! 'course/assessment/question/multiple_responses/switch_question_type_button', locals: {
+  assessment: assessment,
+  question: question,
+  new_question: new_question
+}
+
+json.type question.question_type
```

Comment:
Yes, it was intentional. I needed `_multiple_response_details` to give me the new `question_type` because it is rendered after MCQ/MRQ switch. This new `question_type` is needed to update the question type `Chip` in `Question`.

The perfect solution would be to _not_ have `_question_assessment` render `json.type`, but we create templates for every question type (not just MCQ/MRQ that we have right now), like how we are doing for `_*_conditional`, and let them render their respective `json.type`. This resolves the (a) violation of Law of Demeter in `_question_assessment:11`, and (b) possibility of if-else hell should we wish to do custom partials render for questions other than MCQ/MRQ (e.g., `_question_assessment:29`). But this refactor is probably not really needed now.

Another way is to pass a boolean as one of the `locals` to `_multiple_response_details` to decide to render `json.type` or not. But I think this is unwieldy, and since `_multiple_response_details`'s `json.type` will override that of `_question_assessment`'s, I think this 'overriding' is okay for now.

Let me know if I had missed anything.

Reply:
Got it! Overriding is fine. Could you add a short comment in the codebase as to why the type is needed there? Thanks

Reply:
To escape the if-else issue, you can consider using lookup_context from rails to check if a specific template exists, then render it. (ref https://apidock.com/rails/ActionView/LookupContext/ViewPaths/template_exists%3F)

Reply:
>To escape the if-else issue, you can consider using lookup_context from rails

Thanks for this suggestion. I will keep this in mind should we ever decide to extend the template generation for different question types in `_question_assessment`.